base/c.jl

```diff
@@ -49,7 +50,7 @@ Equivalent to the native `wchar_t` c-type ([`Int32`](@ref)).
 """
 Cwchar_t
 
-if !is_windows()
+if ccall(:jl_get_UNAME, Any, ()) !== :NT
```

Comment:
Hmm, I agree it's very strange that the const version works but without the const it does not.  It sounds like it could be dangerous to go with the workaround unless the issue is  understood.